### PR TITLE
Add multi-ajax

### PR DESCRIPTION
**Pull request by JoseKilo**
Add a MultiAjax object that allows to request several urls by ajax easily.

Every new request is launched after receiving a response to the previous one.
Any error happening in the middle will trigger the error callback and will stop the processing.

### CODE REVIEW

File: src/multi-ajax.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                }
+                if (index === urls.length - 1) {
+                    onload(responses);
+                }
+            }, onerror);
+        },
+        /**
+         * Execute several ajax requests parsing the responses into a list of JSON objects.
+         *
+         * @param {string[]} urls Location of json files
```
Comment (nwhite89) [inline src/multi-ajax.js]: should be array?

File: src/multi-ajax.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    /**
+     * A module to perform several AJAX calls
+     * @exports multi-ajax
+     */
+
+    function MultiAjax () {
+        this.ajax = new Ajax();
+    }
+
+    MultiAjax.prototype._loadJsonCallbackGenerator = function (urls, index, method, onload, onerror, responses) {
```
Comment (nwhite89) [inline src/multi-ajax.js]: can you add JSDoc information about this please

File: src/multi-ajax.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            if (index < urls.length - 1) {
+                $this._loadJson.call($this, urls, index + 1, method, onload, onerror, responses);
+            }
+            if (index === urls.length - 1) {
+                onload(responses);
+            }
+        };
+    };
+
+    MultiAjax.prototype._loadJson = function (urls, index, method, onload, onerror, responses) {
```
Comment (nwhite89) [inline src/multi-ajax.js]: JSDoc added for this also cheers

File: src/multi-ajax.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            urls[index],
+            method,
+            this._loadJsonCallbackGenerator.call(this, urls, index, method, onload, onerror, responses),
+            onerror
+        );
+    };
+
+    /**
+     * Execute several ajax requests parsing the responses into a list of JSON objects.
+     *
```
Comment (nwhite89) [inline src/multi-ajax.js]: needs to have 

```js
* @memberOf module:multi-ajax
* @public
```

File: tests/multi-ajax.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+        it('should be able to initialise', function () {
+            expect(ajax.loadJson).toBeDefined();
+        });
+
+        it('should load json', function () {
+            spyon(multiajax.ajax, 'getrequest').and.callthrough();
+            var result = multiajax.loadjson(['some.json']);
+
+            expect(result).not.tobedefined();
```
Comment (nwhite89) [inline tests/multi-ajax.js]: CamelCase?

File: tests/multi-ajax.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
+define([
```
Comment (nwhite89) [inline tests/multi-ajax.js]: This file should be 

`tests/multi-ajax.spec.js`

File: src/multi-ajax.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    MultiAjax.prototype._loadJsonCallbackGenerator = function (urls, index, method, onload, onerror, responses) {
+        var $this = this;
+        return function (response) {
+            responses.push(response);
+            if (index < urls.length - 1) {
+                $this._loadJson.call($this, urls, index + 1, method, onload, onerror, responses);
+            }
+            if (index === urls.length - 1) {
```
Comment (nwhite89) [inline src/multi-ajax.js]: perhaps this could simply be else ?

File: src/multi-ajax.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    'aux/ajax'
+], function (Ajax) {
+
+    /**
+     * A module to perform several AJAX calls
+     * @exports multi-ajax
+     */
+
+    function MultiAjax () {
+        this.ajax = new Ajax();
```
Comment (nwhite89) [inline src/multi-ajax.js]: should be `this._ajax`

File: tests/multi-ajax.js (hunk 6 above)
Comment (JoseKilo) [inline tests/multi-ajax.js]: renamed

File: src/multi-ajax.js (hunk 2 above)
Comment (JoseKilo) [inline src/multi-ajax.js]: added

File: src/multi-ajax.js (hunk 3 above)
Comment (JoseKilo) [inline src/multi-ajax.js]: added

File: src/multi-ajax.js (hunk 1 above)
Comment (JoseKilo) [inline src/multi-ajax.js]: Seems to be correct, according to --> http://usejsdoc.org/tags-param.html

File: src/multi-ajax.js (hunk 4 above)
Comment (JoseKilo) [inline src/multi-ajax.js]: done

File: src/multi-ajax.js (hunk 8 above)
Comment (JoseKilo) [inline src/multi-ajax.js]: done

File: src/multi-ajax.js (hunk 7 above)
Comment (JoseKilo) [inline src/multi-ajax.js]: done

File: tests/multi-ajax.js (hunk 5 above)
Comment (JoseKilo) [inline tests/multi-ajax.js]: Oops

File: src/multi-ajax.js [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+                $this._loadJson.call($this, urls, index + 1, method, onload, onerror, responses);
+            } else {
+                onload(responses);
+            }
+        };
+    };
+
+    /**
+     * Request and parse a single Json object, using a generated callback to collect the response.
+     *
```
Comment (nwhite89) [inline src/multi-ajax.js]: can you add on `@protected` here please

File: src/multi-ajax.js [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+     * @exports multi-ajax
+     */
+
+    function MultiAjax () {
+        this._ajax = new Ajax();
+    }
+
+    /**
+     * Return a function that can be used as a callback for the Json loading requests.
+     *
```
Comment (nwhite89) [inline src/multi-ajax.js]: can you add on `@protected` here please

File: package.json [hunk 11]
```diff
@@ -35,6 +35,7 @@
     "grunt-karma": "~0.10.1",
     "grunt-karma-coveralls": "~2.5.3",
     "istanbul-instrumenter-loader": "~0.1.2",
+    "jasmine-ajax": "^3.2.0",
```
Comment (nwhite89) [inline package.json]: can you use `~3.2.0` rather than `^` here please

File: src/multi-ajax.js (hunk 9 above)
Comment (JoseKilo) [inline src/multi-ajax.js]: added

File: src/multi-ajax.js (hunk 10 above)
Comment (JoseKilo) [inline src/multi-ajax.js]: added

File: package.json (hunk 11 above)
Comment (JoseKilo) [inline package.json]: done

**Comment by rashmisagar26** [comment]
Passed QA! Ready to merge

**Pull request merged**